tests/backup_helper.go

```diff
@@ -1350,6 +1350,47 @@ func CreateRestoreWithReplacePolicy(restoreName string, backupName string, names
 	return nil
 }
 
+// CreateRestoreWithReplacePolicyWithValidation Creates in-place restore and waits for it to complete and then validates the restore
+func CreateRestoreWithReplacePolicyWithValidation(restoreName string, backupName string, namespaceMapping map[string]string, clusterName string,
+	orgID string, ctx context1.Context, storageClassMapping map[string]string, replacePolicy ReplacePolicyType, scheduledAppContexts []*scheduler.Context) error {
+	err := CreateRestoreWithReplacePolicy(restoreName, backupName, namespaceMapping, clusterName, orgID, ctx, storageClassMapping, replacePolicy)
+	if err != nil {
+		return err
+	}
+	originalClusterConfigPath := CurrentClusterConfigPath
+	if clusterConfigPath, ok := ClusterConfigPathMap[clusterName]; !ok {
+		err = fmt.Errorf("switching cluster context: couldn't find clusterConfigPath for cluster [%s]", clusterName)
+		return err
+	} else {
+		log.InfoD("Switching cluster context to cluster [%s]", clusterName)
+		err = SetClusterContext(clusterConfigPath)
+		if err != nil {
+			return err
+		}
+	}
+	defer func() {
+		log.InfoD("Switching cluster context back to cluster path [%s]", originalClusterConfigPath)
+		err := SetClusterContext(originalClusterConfigPath)
+		if err != nil {
+			log.FailOnError(err, "Failed switching cluster context back to cluster path [%s]", originalClusterConfigPath)
```

Comment:
How about we name the return values and assign the error?

```go
// CreateRestoreWithReplacePolicyWithValidation Creates in-place restore and waits for it to complete and then validates the restore
func CreateRestoreWithReplacePolicyWithValidation(restoreName string, backupName string, namespaceMapping map[string]string, clusterName string, orgID string, ctx context1.Context, storageClassMapping map[string]string, replacePolicy ReplacePolicyType, scheduledAppContexts []*scheduler.Context) (err error) {
```

```go
defer func() {
	log.InfoD("Switching cluster context back to cluster path [%s]", originalClusterConfigPath)
	err := SetClusterContext(originalClusterConfigPath)
	if err != nil {
		err = fmt.Errorf("Failed switching cluster context back to cluster path [%s]. Err: [%v]", originalClusterConfigPath, err)
	}
}()
```

Reply:
Changed it

```
// CreateRestoreWithReplacePolicyWithValidation Creates in-place restore and waits for it to complete and then validates the restore
func CreateRestoreWithReplacePolicyWithValidation(restoreName string, backupName string, namespaceMapping map[string]string, clusterName string,
	orgID string, ctx context1.Context, storageClassMapping map[string]string, replacePolicy ReplacePolicyType, scheduledAppContexts []*scheduler.Context) (err error) {
	err = CreateRestoreWithReplacePolicy(restoreName, backupName, namespaceMapping, clusterName, orgID, ctx, storageClassMapping, replacePolicy)
	if err != nil {
		return
	}
	originalClusterConfigPath := CurrentClusterConfigPath
	if clusterConfigPath, ok := ClusterConfigPathMap[clusterName]; !ok {
		err = fmt.Errorf("switching cluster context: couldn't find clusterConfigPath for cluster [%s]", clusterName)
		return
	} else {
		log.InfoD("Switching cluster context to cluster [%s]", clusterName)
		err = SetClusterContext(clusterConfigPath)
		if err != nil {
			return
		}
	}
	defer func() {
		log.InfoD("Switching cluster context back to cluster path [%s]", originalClusterConfigPath)
		err = SetClusterContext(originalClusterConfigPath)
		if err != nil {
			err = fmt.Errorf("failed switching cluster context back to cluster path [%s]. Err: [%v]", originalClusterConfigPath, err)
		}
	}()
	expectedRestoredAppContexts := make([]*scheduler.Context, 0)
	for _, scheduledAppContext := range scheduledAppContexts {
		expectedRestoredAppContext, err := CloneAppContextAndTransformWithMappings(scheduledAppContext, namespaceMapping, storageClassMapping, true)
		if err != nil {
			log.Errorf("TransformAppContextWithMappings: %v", err)
			continue
		}
		expectedRestoredAppContexts = append(expectedRestoredAppContexts, expectedRestoredAppContext)
	}
	err = ValidateRestore(ctx, restoreName, orgID, expectedRestoredAppContexts, make([]string, 0))
	if err != nil {
		return
	}
	return
}
```